packages/block-library/src/embed/core-embeds.js

```diff
@@ -354,6 +354,32 @@ export const others = [
 			icon: embedVideoIcon,
 			keywords: [ __( 'video' ) ],
 			description: __( 'Embed a VideoPress video.' ),
+			transforms: {
+				from: [
+					{
+						type: 'raw',
+						isMatch: ( node ) => {
+							if ( node.nodeName !== 'P' ) {
+								return false;
+							}
+
+							if ( node.childNodes.length !== 1 || node.childNodes[ 0 ].nodeName !== 'EMBED' ) {
```

Comment:
Not sure if the embed is expected to always have the same elements, but an alternative here and on line 374 could be to use `node.querySelector( 'embed' )` to check for the presence of the embed tag.

Reply:
What if there's something like `<p>some text <embed></p>`?